src/python_inspector/setup_py_live_eval.py

```diff
@@ -0,0 +1,188 @@
+    """Parse pip requirements file."""
+    # requirement lines sorted by importance
+    # also collect other pip commands
+    rdev = {}
+    rnormal = []
+    stuff = []
+
+    try:
+        with open(path) as f:
+            for line in f:
+                line = line.strip()
+
+                # see https://pip.readthedocs.io/en/1.1/requirements.html
+                if line.startswith("-e"):
+                    # devel requirement
+                    splitted = line.split("#egg=")
+                    rdev[splitted[1].lower()] = line
+
+                elif line.startswith("-r"):
+                    # recursive file command
+                    splitted = re.split("-r\\s+", line)
+                    subrdev, subrnormal, substuff = parse_pip_file(
+                        os.path.join(os.path.dirname(path), splitted[1])
+                    )
+                    for k, v in subrdev.items():
+                        if k not in rdev:
+                            rdev[k] = v
+                    rnormal.extend(subrnormal)
+                elif line.startswith("-"):
+                    # another special command we don't recognize
+                    stuff.append(line)
+                else:
+                    # ordinary requirement, similarly to them used in setup.py
+                    rnormal.append(line)
+    except IOError:
+        print('Warning: could not parse requirements file "{0}"!'.format(path), file=sys.stderr)
+
+    return rdev, rnormal, stuff
+
+
+def iter_requirements(level, extras, setup_file):
+    """Iterate over requirements."""
+    from pathlib import Path
+
+    setup_file = str(Path(setup_file).absolute())
+    result = dict()
+    requires = []
+    stuff = []
+    cd = os.getcwd()
+    os.chdir(os.path.dirname(setup_file))
+    install_requires = []
+    requires_extras = {}
+    # change directory to setup.py path
+    with mock.patch.object(setuptools, "setup") as mock_setup:
+        sys.path.append(os.path.dirname(setup_file))
+        g = {"__file__": setup_file, "__name__": "__main__"}
+        with open(setup_file) as sf:
+            exec(sf.read(), g)
+        sys.path.pop()
+        assert g["setup"]  # silence warning about unused imports
+    # called arguments are in `mock_setup.call_args`
+    os.chdir(cd)
+    mock_args, mock_kwargs = mock_setup.call_args
+    install_requires = mock_kwargs.get("install_requires", install_requires)
+
+    requires_extras = mock_kwargs.get("extras_require", requires_extras)
+
+    for e, reqs in requires_extras.items():
+        # Handle conditions on extras. See pkginfo_to_metadata function
+        # in Wheel for details.
+        condition = ""
+        if ":" in e:
+            e, condition = e.split(":", 1)
+        if not e or e in extras:
+            if condition:
+                reqs = ["{0}; {1}".format(r, condition) for r in reqs]
+            install_requires.extend(reqs)
+
+    for pkg in pkg_resources.parse_requirements(install_requires):
```

Comment:
This likely is worth replacing by our own requirements parsing which is better (or at least use "packaging" and not pkg_resources).